modules/core/src/v2/coins/eth.ts

```diff
@@ -1196,6 +1208,119 @@ export class Eth extends BaseCoin {
+      throw new InvalidEthereumBatchData('batch data function signature hash does not match expected');
+    }
+
+    const [addresses, amounts] = optionalDeps.ethAbi.rawDecode(
+      Eth.batchMethodTypes,
+      Buffer.from(batchData.slice(10), 'hex')
+    );
+    if (addresses.length !== amounts.length) {
+      throw new InvalidEthereumBatchData('recipients and amounts arrays must be the same size');
+    }
+
+    const batchRecipients = [];
+    let totalAmount = new BigNumber(0);
+    for (let i = 0; i < addresses.length; i++) {
+      const address = addresses[i].toString('hex');
+      if (!this.isValidAddress(address)) {
+        throw new InvalidAddressError('invalid address: ' + address);
+      }
+
+      const amount = amounts[i].toString();
+      if (typeof amount !== 'string' || !amount.match(/^[0-9]+$/)) {
+        throw new InvalidAmountError('invalid amount: ' + amounts[i].toString());
+      }
+      totalAmount = totalAmount.plus(amount);
+
+      batchRecipients.push({
+        address,
+        amount,
+      });
+    }
+
+    const sortedBatchRecipients = _(batchRecipients)
+      .sortBy(['address', 'amount'])
+      .map(({ address, amount }) => ({
+        address: optionalDeps.ethUtil.addHexPrefix(address),
+        amount: new BigNumber(amount).toFixed(),
+      }))
+      .value();
+
+    const sortedOriginalRecipients = _(originalRecipients)
+      .sortBy(['address', 'amount'])
+      .map(({ address, amount }) => ({
+        address: optionalDeps.ethUtil.addHexPrefix(address),
+        amount: new BigNumber(amount).toFixed(),
+      }))
+      .value();
+
+    if (!_.isEqual(sortedBatchRecipients, sortedOriginalRecipients)) {
+      throw new InvalidEthereumBatchData('batch recipients in data does not match original recipients');
+    }
+
+    if (!totalAmount.eq(sendAmount)) {
+      throw new InvalidEthereumBatchData(
+        `Batcher send amount ${sendAmount} is not equal to sum of individual transfers ${totalAmount.toFixed()}`
+      );
+    }
+  }
+
+  /**
+   * Validates that the prebuild data returned by platform is valid
+   * @param params
+   * @param callback
+   * @throws Error if the prebuild is invalid
+   */
+  verifyTransaction(params: any, callback?: NodeCallback<any>): Bluebird<any> {
+    return co(function*() {
+      const { txParams, txPrebuild } = params;
+      if (txPrebuild.recipients.length !== 1) {
+        throw new InvalidEthereumRecipients(
+          `prebuild recipients array should always be of length 1, was of length ${txPrebuild.recipients.length} here`
+        );
+      }
+      if (txParams.recipients.length === 1 && !_.isEqual(txPrebuild.recipients, txParams.recipients)) {
+        throw new InvalidEthereumRecipients(
+          'recipients returned by BitGo are not equal to recipients in original tx request'
+        );
+      }
+      if (txParams.recipients.length > 1) {
+        this.validateBatchData(txPrebuild.recipients, txParams.recipients);
```

Comment:
Typescript has trouble understanding the type of `this` inside of coroutines (long story). To help it out, we usually declare a new variable `self` and assign it to `this` at the  start of the function, then  use `self` throughout.  This helps the typescript compiler  check types on calls involving `this`, which it otherwise couldn't do.

TL;DR: this should be `self.validateBatchData()`.

Reply:
Fixed.